### ISSUE TITLE
Doxygen member groupings formatted incorrectly

### ISSUE THREAD

**Issue by ischoegl** [Cantera/cantera#1275]
**Problem description**

Among other instances, the following documentation
https://github.com/Cantera/cantera/blob/2bc02637b1d6df0fb4203b70c4efafe14e0fb9fc/include/cantera/kinetics/Kinetics.h#L1190-L1208
does not produce the expected output in the [doxygen documentation](https://cantera.org/documentation/docs-2.6/doxygen/html/d4/dc4/classCantera_1_1Kinetics.html#a562bd1bb719af977b3eb0c35ea482d92). 

**Behavior**

Presumbably, the `* @{ ` starts a new section that includes multiple methods (ending with the next `//! @}`). Instead, the section description is merged with the first method listed.

**System information**

- Cantera version: Cantera 2.6
- OS: N/A (online documentation)

**Comment by bryanwweber** [comment]
Can you note what Doxygen version you're using?

**Comment by ischoegl** [comment]
These are the official documents (online Cantera documentation), so it’s not something I produced locally.

**PS:** I actually don't think that it is a rendering issue, so I updated the issue title.

**Comment by ischoegl** [comment]
@bryanwweber ... the GH action builds on `ubuntu-latest`, where the native doxygen per [package info](https://packages.ubuntu.com/focal/doxygen) is 1.8.17.

**Comment by ischoegl** [comment]
Per documentation, this appears to be a case of a **Member Grouping**, see [doxygen manual](https://www.doxygen.nl/manual/grouping.html#memgroup) (as opposed to [Modules](https://www.doxygen.nl/manual/grouping.html#modules)). Instructions indicate that the correct syntax should be
```
///@{ 
  ...
///@}
```
or
```
/**@{*/ 
  ... 
/**@}*/ 
```

**Comment by ischoegl** [comment]
I ran some tests and found the following: while this works (despite a slightly different commenting style)
```C++
    //! @name Altering Reaction Rates
    //! These methods alter reaction rates. They are designed primarily for
    //! carrying out sensitivity analysis, but may be used for any purpose
    //! requiring dynamic alteration of rate constants. For each reaction, a
    //! real-valued multiplier may be defined that multiplies the reaction rate
    //! coefficient. The multiplier may be set to zero to completely remove a
    //! reaction from the mechanism.
    //! @{
    [...]
    //! @}
```
the following does not
```C++
    //! @name Altering Reaction Rates
    /*!
     * These methods alter reaction rates. They are designed primarily for
     * carrying out sensitivity analysis, but may be used for any purpose
     * requiring dynamic alteration of rate constants. For each reaction, a
     * real-valued multiplier may be defined that multiplies the reaction rate
     * coefficient. The multiplier may be set to zero to completely remove a
     * reaction from the mechanism.
     * @{
     */
    [...]
    //! @}
```
In particular, it appears to be important that the commenting style is consistent; mixing different styles produces unexpected outcomes (even if the grouping is started with `//! @{` while leaving the other comments unchanged). ~I added fixes for the `include/cantera/kinetics` folder in #1274~ ... separated out as #1276

**PS:** The following also works
```C++
    /**
     * @}
     * @name Activities, Standard States, and Activity Concentrations
     *
     * The activity \f$a_k\f$ of a species in solution is related to the
     * chemical potential by \f[ \mu_k = \mu_k^0(T) + \hat R T \log a_k. \f] The
     * quantity \f$\mu_k^0(T,P)\f$ is the chemical potential at unit activity,
     * which depends only on temperature and pressure.
     * @{
     */
```

**Comment by speth** [comment]
Thanks for catching this, @ischoegl. I had fixed some other issues related to these grouping commands previously (#1152), but hadn't noticed that the extended description of the group was occasionally ending being misinterpreted.

The apparent requirement here is that the extended description has to be in the same "Doxygen comment block" as the `@name` directive that specifies the heading, and the reason we're seeing issues is that this is seen as two comment blocks:
```c++
    //! @}
    //! @name Electric Potential
    /*! 
     *  The phase may be at some non-zero electrical potential. These methods
     *  set or get the value of the electric potential.
     *  @{
     */
```

Essentially as if it had been written:
```c++
    //! @}
    //! @name Electric Potential

    //! The phase may be at some non-zero electrical potential. These methods
    //! set or get the value of the electric potential.
    //! @{
```

As far as I can tell, it doesn't matter which commenting style or block the grouping delimiters are in, as long as they are in a Doxygen comment block of one sort or another.

**Comment by ischoegl** [comment]
Correct - this is exactly my interpretation. For the fix in #1276, I defaulted to one consistent formatting style for member groupings. I left module groupings alone.

I used the C++ style as this appears to be more consistently used in newer additions (e.g. `AnyValue`). Also, the `/** @}*/` is a little harsher than the `\\! @}` that had been widely introduced some time ago.
